src/main/java/com/redhat/devtools/lsp4ij/features/formatting/LSPEditorImprovementsEnterBetweenBracesHandler.java

```diff
@@ -0,0 +1,200 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.features.formatting;
+
+import com.intellij.application.options.CodeStyle;
+import com.intellij.codeInsight.CodeInsightSettings;
+import com.intellij.codeInsight.editorActions.enter.EnterBetweenBracesHandler;
+import com.intellij.openapi.actionSystem.DataContext;
+import com.intellij.openapi.editor.CaretModel;
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.editor.actionSystem.EditorActionHandler;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.Ref;
+import com.intellij.openapi.util.text.StringUtil;
+import com.intellij.psi.PsiDocumentManager;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.codeStyle.CodeStyleSettings;
+import com.redhat.devtools.lsp4ij.LSPIJEditorUtils;
+import com.redhat.devtools.lsp4ij.client.features.LSPEditorBehaviorFeature;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * This provides a workaround for <a href="https://youtrack.jetbrains.com/issue/IJPL-159454">IJPL-159454</a>.
+ */
+public class LSPEditorImprovementsEnterBetweenBracesHandler extends EnterBetweenBracesHandler {
+    private boolean enterAfterOpenBrace = false;
+    private Character openBraceCharacter = null;
+    private boolean enterBeforeCloseBrace = false;
+    private Character closeBraceCharacter = null;
+
+    @Override
+    public Result preprocessEnter(
+            @NotNull PsiFile file,
+            @NotNull Editor editor,
+            @NotNull Ref<Integer> caretOffset,
+            @NotNull Ref<Integer> caretAdvance,
+            @NotNull DataContext dataContext,
+            @Nullable EditorActionHandler originalHandler) {
+        if (LSPIJEditorUtils.isSupportedTextMateFile(file) &&
+            CodeInsightSettings.getInstance().SMART_INDENT_ON_ENTER &&
+            LSPEditorBehaviorFeature.enableTextMateEnterBetweenBracesFix(file)) {
```

Comment:
Immediate feature flag check so if this feature is disabled in client config, this is a no-op.